### ISSUE TITLE
Window not drawing correctly.

### ISSUE THREAD

**Issue by cinjon** [syl20bnr/spacemacs#14306]
#### Description :octocat:
On a fresh install of emacs spacemacs (on mac os), whenever I open a split pane, the new window doesn't paint correctly. See below for a screenshot. This also happens in other situations, like if I do C-l to recenter.

#### Reproduction guide :beetle:
- Start Emacs
- Open a new pane with C-x 3 for example and the pane is drawn badly.

*Observed behaviour:* :eyes: :broken_heart:
The observed behavior is that the pane is drawn really badly. It doesn't even always get fixed with redraw-display, like for example in this window while I'm filing the bug.

*Expected behaviour:* :heart: :smile:
The expected behavior is that it paints properly. I didn't have this issue on my previous computer at all.

#### System Info :computer:
- OS: darwin
- Emacs: 27.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. b2f014107)
- Graphic display: nil
- Distribution: spacemacs
- Editing style: emacs
- Completion: ivy
- Layers:
```elisp
(emacs-lisp git helm python ivy multiple-cursors treemacs)
```
- System configuration features: RSVG IMAGEMAGICK GLIB NOTIFY KQUEUE ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES THREADS JSON PDUMPER LCMS2 GMP

![Screen Shot 2021-01-19 at 8 32 46 AM](https://user-images.githubusercontent.com/615351/105042142-e06bf180-5a31-11eb-912f-1ea1865c3494.png)
![Screen Shot 2021-01-19 at 8 39 02 AM](https://user-images.githubusercontent.com/615351/105042151-e366e200-5a31-11eb-8d38-eaa944a4e082.png)

**Comment by lebensterben** [comment]
what's the font. and what's the terminal emulator you're using

**Comment by cinjon** [comment]
This is on Mac's default Terminal using Solarized Dark with Menlo Regular 11pt.

**Comment by lebensterben** [comment]
does the problem also exist in GUI Emacs

**Comment by cinjon** [comment]
From a minute of trying stuff, it doesn't seem to be an issue there.

**Comment by lebensterben** [comment]
that's what I suspected. there's something wrong with your terminal setting. from the pictures attached above, it seems that anti-aliasing is not working.
Note that the font used by terminal Emacs is not what you defined in dotspacemacs but the font use for terminal.

**Comment by cinjon** [comment]
Thanks, that's helpful. However, I'm having a bit of trouble squaring what is exactly to fix. Are you suggesting aligning the fonts in dotspacemacs with what's used in terminal or changing a setting (somewhere) or something else? I tried a bunch of the suggestions here without success (https://github.com/syl20bnr/spacemacs/issues/1269). I also checked and "Antialias text" is on by default in the terminal profile. 

Is there an expectation that terminal is xterm? Right now, it's being declared as ansi.
